### PR TITLE
Fixes 2048,2613,1700: React 18 preview

### PR DESCRIPTION
**Pull request by Andrewgdewar**
## Summary

This is the ongoing react 18 upgraded "main" branch, created to hold acked changes until IQE-core/QE have Patternfly 5/React 18 compatibility for their test suites.

This can be merged to main once all tests are passing and QE is confident of test compatibility.

## Testing steps

- None at the moment.

### CODE REVIEW

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMS-2048

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMS-Fixes

**Comment by swadeley** [comment]
Hi

<h1 data-ouia-component-type="PF4/Title" data-ouia-safe="true" data-ouia-component-id="OUIA-Generated-Title-1" widget-type="InsightsPageHeaderTitle" class="pf-c-title pf-m-2xl">Custom Repositories</h1>

 why that says data-ouia-component-type="PF4/Title"

**Comment by swadeley** [comment]
Hi

this Kebab section is quite different to the old version:
`
<button class="pf-v5-c-menu-toggle pf-m-plain" type="button" aria-label="Kebab toggle" aria-expanded="false">`

Try add `data-ouia-component-id='custom_repositories_kebab_toggle'

Thank you

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMS-1700

File: .github/workflows/build-repo.yml [hunk 1]
```diff
@@ -6,9 +6,6 @@ on:
 jobs:
   build-test-release:
     runs-on: ubuntu-latest
-    strategy:
```
Comment (mshriver) [inline .github/workflows/build-repo.yml:9]: With this we lose the ability to specify a future version for testing/building before we try to move to it.

Worth the tradeoff to use the `.nvmrc` directly here? It is really nice to not duplicate the 'current' version in the matrix.

Comment (Andrewgdewar) [reply .github/workflows/build-repo.yml:9]: More up to you guys. It doesn't effect my workflow (local) either way. 
If it's a change you'd like reverted I a-okay with that.

Comment (Andrewgdewar) [reply .github/workflows/build-repo.yml:9]: @mshriver

Comment (Andrewgdewar) [reply .github/workflows/build-repo.yml:9]: For now we'll leave it as is, if you wish to change it in the future we can do that quite easily.

Comment (mshriver) [reply .github/workflows/build-repo.yml:9]: Missed your response - not taking a hard stance here, just thought it was worth consideration. I doubt we'll be moving node versions enough to make it necessary.

Agreed, if we find ourselves wanting next version results early we can modify without much trouble.

**Comment by swadeley** [comment]
Hi, the updated-for-Patternfly 5 tests pass. Lets merge this.

**Review by rverdile — APPROVED** [review]
lgtm!

**Review by jlsherrill — APPROVED** [review]
I approve too!

**Comment by swadeley** [comment]
OK, lgtm, merge when test status permits please.

**Pull request merged**